xbmc/platform/android/activity/XBMCApp.cpp

```diff
@@ -511,35 +510,25 @@ void CXBMCApp::Quit()
   android_printf("%s: Application stopped!", __PRETTY_FUNCTION__);
 }
 
-bool CXBMCApp::EnableWakeLock(bool on)
+void CXBMCApp::KeepScreenOnCallback(void* onVariant)
 {
-  android_printf("%s: %s", __PRETTY_FUNCTION__, on ? "true" : "false");
-  if (!m_wakeLock)
-  {
-    std::string appName = CCompileInfo::GetAppName();
-    StringUtils::ToLower(appName);
-    std::string className = CCompileInfo::GetPackage();
-    // SCREEN_BRIGHT_WAKE_LOCK is marked as deprecated but there is no real alternatives for now
-    m_wakeLock =
-        std::make_unique<CJNIWakeLock>(CJNIPowerManager(getSystemService("power"))
-                                           .newWakeLock(CJNIPowerManager::SCREEN_BRIGHT_WAKE_LOCK |
-                                                            CJNIPowerManager::ON_AFTER_RELEASE,
-                                                        className));
-    m_wakeLock->setReferenceCounted(false);
-  }
+  CVariant* onV = static_cast<CVariant*>(onVariant);
+  bool on = onV->asBoolean();
+  delete onV;
 
-  if (on)
-  {
-    if (!m_wakeLock->isHeld())
-      m_wakeLock->acquire();
-  }
-  else
+  CJNIWindow window = getWindow();
+  if (window)
   {
-    if (m_wakeLock->isHeld())
-      m_wakeLock->release();
+    on ? window.addFlags(CJNIWindowManagerLayoutParams::FLAG_KEEP_SCREEN_ON)
+       : window.clearFlags(CJNIWindowManagerLayoutParams::FLAG_KEEP_SCREEN_ON);
   }
+}
 
-  return true;
+void CXBMCApp::KeepScreenOn(bool on)
+{
+  android_printf("%s: %s", __PRETTY_FUNCTION__, on ? "true" : "false");
+  CVariant* variant = new CVariant(on);
```

Comment:
Maybe add a comment that this is cleaned up in the callback and the callback is guaranteed to be called at least once to prevent a mem leak.

Reply:
ok, add a comment to inform that the object is properly deallocated